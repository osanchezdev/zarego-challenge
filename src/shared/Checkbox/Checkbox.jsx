import React from 'react';
import { bool, func, string } from 'prop-types';
import styles from './Checkbox.module.css';

const Checkbox = ({ name, label, showLabel, controller }) => (
  <div className={styles.checkbox__wrapper}>
    <div className={styles.checkbox_input__wrapper}>
      <input
        id={name}
        type="checkbox"
        name={name}
        className={styles.checkbox_input}
        {...controller(name)}
      />
    </div>
    {label && showLabel && (
      <div className={styles.checkbox_label__wrapper}>
        <label htmlFor={name} className={styles.checkbox_label}>
          {label}
        </label>
      </div>
    )}
  </div>
);
Checkbox.propTypes = {
  label: string.isRequired,
  name: string.isRequired,
  showLabel: bool,
  controller: func.isRequired,
};

Checkbox.defaultProps = { showLabel: true };

export default Checkbox;
