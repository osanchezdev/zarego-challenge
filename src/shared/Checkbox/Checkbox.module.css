.checkbox__wrapper {
  display: flex;
  align-items: center;
}

.checkbox_input__wrapper {
  position: relative;
}

.checkbox_input {
  cursor: pointer;
  height: 25px;
  width: 25px;
}

.checkbox_label__wrapper {
  margin-left: 10px;
}

.checkbox_label {
  position: relative;
}
