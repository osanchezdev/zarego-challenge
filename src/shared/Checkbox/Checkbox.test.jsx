import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

const { labelText, name, controller } = {
  labelText: 'labelTest',
  name: 'fieldName',
  controller: jest.fn(),
};
describe('Checkbox component snapshot', () => {
  it('should render component', () => {
    const tree = renderer
      .create(
        <Checkbox label={labelText} name={name} controller={controller} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Checkbox component funtional tests', () => {
  it('should not be checked', () => {
    render(<Checkbox label={labelText} name={name} controller={controller} />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).not.toBeChecked();
  });
});
