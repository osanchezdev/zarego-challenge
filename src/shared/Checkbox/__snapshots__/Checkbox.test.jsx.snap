// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component snapshot should render component 1`] = `
<div
  className="checkbox__wrapper"
>
  <div
    className="checkbox_input__wrapper"
  >
    <input
      className="checkbox_input"
      id="fieldName"
      name="fieldName"
      type="checkbox"
    />
  </div>
  <div
    className="checkbox_label__wrapper"
  >
    <label
      className="checkbox_label"
      htmlFor="fieldName"
    >
      labelTest
    </label>
  </div>
</div>
`;
